lib/invoice.rb

```diff
@@ -9,10 +9,10 @@ class Invoice
 
   def initialize(data)
     @id          = data[:id].to_i
-    @customer_id = data[:customer_id]
-    @merchant_id = data[:merchant_id]
-    @status      = data[:status]
-    @created_at  = data[:created_at]
-    @updated_at  = data[:updated_at]
+    @customer_id = data[:customer_id].to_i
+    @merchant_id = data[:merchant_id].to_i
+    @status      = data[:status].to_sym
+    @created_at  = Time.parse(data[:created_at])
+    @updated_at  = Time.parse(data[:updated_at])
   end
```

Comment:
nice type conversions here. had to update the merchant_id on my branch to get sales_analyst running so we may see a small conflict when I merge but we'll want to keep things the way you have them here :)